lib/puppet/type/file/content.rb

```diff
@@ -119,6 +120,11 @@ def insync?(is)
       result
     end
 
+    def property_matches?(current, desired)
+      return super unless [:mtime, :ctime].include?(checksum_type)
+      DateTime.parse(current) >= DateTime.parse(desired)
```

Comment:
mtime/ctime/Time.now all return the format "2015-01-29 11:40:47 -0800" on both Windows and Linux. DateTime.parse is also pretty flexible about parsing.